python/ray/data/_internal/planner/plan_read_op.py

```diff
@@ -105,7 +105,10 @@ def get_input_data() -> List[RefBundle]:
                         cleaned_metadata(read_task),
                     )
                 ],
-                owns_blocks=True,
+                # `owns_blocks` is False, because these refs are the root of the DAG.
+                # We shouldn't eagerly free them. Otherwise, the DAG cannot be
+                # reconstructed.
+                owns_blocks=False,
```

Comment:
should we also change of legacy compat code path - https://github.com/ray-project/ray/blob/master/python/ray/data/_internal/execution/legacy_compat.py#L223 ?